ap/tasks/histograms.py

```diff
@@ -27,22 +27,24 @@ class CreateHistograms(
 
     sandbox = dev_sandbox("bash::$AP_BASE/sandboxes/venv_columnar.sh")
 
-    shifts = {ReduceEvents}
+    shifts = {MergeReducedEvents}
 
     def workflow_requires(self):
         reqs = super(CreateHistograms, self).workflow_requires()
-        if not self.pilot:
-            reqs["events"] = ReduceEvents.req(self)
-            if self.producers:
-                reqs["producers"] = [ProduceColumns.req(self, producer=p) for p in self.producers]
+
+        reqs["events"] = MergeReducedEvents.req(self)
+        if not self.pilot and self.producers:
+            reqs["producers"] = [ProduceColumns.req(self, producer=p) for p in self.producers]
+
         return reqs
 
     def requires(self):
-        reqs = {"events": ReduceEvents.req(self)}
+        reqs = {"events": MergeReducedEvents.req(self, tree_index=self.branch, _exclude={"branch"})}
```

Comment:
I thought about that again. Since at this point one is dealing with a branch task and not the workflow, parameters that are targetting the functionality of the workflow are anyway not passed in req. However, "branches" should be excluded in the "workflow_requires". I will add that in the next commit.